Kernel/Arch/aarch64/RPi/GPIO.cpp

```diff
@@ -94,4 +94,26 @@ void GPIO::internal_enable_pins(u32 enable[2], PullUpDownState state)
     // I don't know if the RPi3 has that already, so this uses the old BCM2835 approach for now.
 }
 
+void GPIO::set_pin_high_detect_enable(unsigned pin_number, bool enable)
+{
+    u32 data[2] = {};
```

Comment:
What's the advantage of loading both words into a local array? Doesn't

```
    if (enable) {
        if (pin_number < 32)
            m_registers->high_detect_enable.bits[0] |= (1 << pin_number);
        else
            m_registers->high_detect_enable.bits[1] |= (1 << (pin_number - 32));
    } else {
        if (pin_number < 32)
            m_registers->high_detect_enable.bits[0] &= ~(1 << pin_number);
        else
            m_registers->high_detect_enable.bits[1] &= ~(1 << (pin_number - 32));
    }
```

do the same thing with less code?

Reply:
No advantage, this was code from [FireFox317](https://github.com/FireFox317) I mindlessly copy-pasted.

I've changed it to yours.